syntax = "proto3";
package eqproto;

message ChannelMessage {
  int32 chan_num = 1;
  int32 language = 2;
  string from = 3;
  string to = 4;
  string message = 5;
  int32 guilddbid = 6;
  string deliverto = 7;
  int32 type = 8;
  int32 minstatus = 9;
  int32 fromadmin = 10;
  bool noreply = 11;
  bool is_emote = 12;
  //0 not queued, 1 queued, 2 queue full, 3 offline
  int32 queued = 13;
  //You can specify a zone id if you want a message exclusively to one zone
  int32 zone_id = 14;
}


//Daily Gain is a special system for tracking players progression in a daily snapshot.
message DailyGain  {
  int32 account_id = 1;
  int32 character_id  = 2;
  int32 levels_gained = 3;
  int32 experience_gained = 4;
  int32 money_earned = 5;  
  string identity = 6;
}